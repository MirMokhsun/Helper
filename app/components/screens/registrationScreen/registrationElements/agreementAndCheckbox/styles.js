import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        width: 290,
        marginBottom: 25,
        // backgroundColor:'red'
    },
    link: {
        textDecorationLine: 'underline',
        textAlign: 'left'
    },
    text: {
        color: '#9FA0B2',
        fontSize: 12,
        fontFamily: 'Roboto-Light',
        lineHeight: 14,
        textAlignVertical: 'bottom',
        marginLeft: 10,
        textAlign: 'left',
    },
    linkwrapper: {
        width: 260,
        flexDirection: 'column',
    },
    checkBox: {
        width: 20,
        height: 20,
        marginHorizontal: 5,
        backgroundColor: '#fff',
        justifyContent: 'center',
        alignItems: 'center',
        borderColor: '#9FA0B2',
        borderWidth: 0.5
    },
    linkedTextWrapper: {
        width: 260,
    },
});