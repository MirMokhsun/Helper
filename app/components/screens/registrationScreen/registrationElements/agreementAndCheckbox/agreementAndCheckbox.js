import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { styles } from './styles';
import { withNavigation } from 'react-navigation';
import Checked from '../../../../commonComponents/svg/Checked';

export class AgreementAndCheckbox extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            showCheck: false
        }
    }

    UNSAFE_componentWillReceiveProps({ isChecked }) {
        let result = null;
        if (isChecked) {
            result = isChecked ? <Checked /> : null;
        }
        this.setState({ showCheck: result });
    }

    routeAgreementScreen = () => {
        const { navigation, title } = this.props;
        navigation.navigate('AgreementScreen', { title });
    }

    render() {
        const { setCheckedBox, textWithClick, text } = this.props;
        const { showCheck } = this.state;
        return (
            <View style={styles.container}>
                <TouchableOpacity onPress={setCheckedBox} testID="BtnCheckBoxRegistrationID" accessibilityLabel="BtnCheckBoxRegistrationID">
                    <View style={styles.checkBox}>
                        {showCheck}
                    </View>
                </TouchableOpacity >
                <View style={styles.linkwrapper}>
                    <Text style={styles.text}>{text} </Text>
                    <TouchableOpacity onPress={this.routeAgreementScreen} testID="BtnAgreementRegistrationID" accessibilityLabel="BtnAgreementRegistrationID">
                        <View style={styles.linkedTextWrapper}>
                            <Text style={[styles.link, styles.text]}>{textWithClick} </Text>
                        </View>
                    </TouchableOpacity  >
                </View>
            </View>
        )
    }
}

export default withNavigation(AgreementAndCheckbox);