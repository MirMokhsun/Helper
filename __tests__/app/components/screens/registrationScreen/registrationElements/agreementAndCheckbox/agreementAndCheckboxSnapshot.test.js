import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AgreementAndCheckbox } from '../../../../../../../app/components/screens/registrationScreen/registrationElements/agreementAndCheckbox/agreementAndCheckbox';

it('create snapshot AgreementAndCheckbox', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<AgreementAndCheckbox
        navigation={() => { }}
        isChecked={true}
        setCheckedBox={() => { }}
        textWithClick={"user agreement"}
        text={"I have read and agree with"}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});