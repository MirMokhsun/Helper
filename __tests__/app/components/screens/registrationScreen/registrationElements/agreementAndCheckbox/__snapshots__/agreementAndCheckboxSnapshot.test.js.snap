// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot AgreementAndCheckbox 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 25,
      "width": 290,
    }
  }
>
  <TouchableOpacity
    accessibilityLabel="BtnCheckBoxRegistrationID"
    activeOpacity={0.2}
    onPress={[Function]}
    testID="BtnCheckBoxRegistrationID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#9FA0B2",
          "borderWidth": 0.5,
          "height": 20,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "width": 20,
        }
      }
    />
  </TouchableOpacity>
  <View
    style={
      Object {
        "flexDirection": "column",
        "width": 260,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9FA0B2",
          "fontFamily": "Roboto-Light",
          "fontSize": 12,
          "lineHeight": 14,
          "marginLeft": 10,
          "textAlign": "left",
          "textAlignVertical": "bottom",
        }
      }
    >
      I have read and agree with
       
    </Text>
    <TouchableOpacity
      accessibilityLabel="BtnAgreementRegistrationID"
      activeOpacity={0.2}
      onPress={[Function]}
      testID="BtnAgreementRegistrationID"
    >
      <View
        style={
          Object {
            "width": 260,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
                "textDecorationLine": "underline",
              },
              Object {
                "color": "#9FA0B2",
                "fontFamily": "Roboto-Light",
                "fontSize": 12,
                "lineHeight": 14,
                "marginLeft": 10,
                "textAlign": "left",
                "textAlignVertical": "bottom",
              },
            ]
          }
        >
          user agreement
           
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
